test/controllers/update/due.diligence.overseas.entity.controller.spec.ts

```diff
@@ -0,0 +1,563 @@
+import { authentication } from "../../../src/middleware/authentication.middleware";
+import { ApplicationDataType } from '../../../src/model';
+import { EMAIL_ADDRESS } from "../../__mocks__/session.mock";
+import { EMPTY_IDENTITY_DATE_REQ_BODY_MOCK, getTwoMonthOldDate } from "../../__mocks__/fields/date.mock";
+import { ErrorMessages } from '../../../src/validation/error.messages';
+import { DateTime } from "luxon";
+import { OVERSEAS_ENTITY_DUE_DILIGENCE_WITH_INVALID_CHARACTERS_FIELDS_MOCK } from "../../__mocks__/validation.mock";
+import {
+  DUE_DILIGENCE_REQ_BODY_OBJECT_MOCK,
+  DUE_DILIGENCE_REQ_BODY_OBJECT_MOCK_FOR_IDENTITY_DATE
+} from "../../__mocks__/due.diligence.mock";
+import {
+  OVERSEAS_ENTITY_DUE_DILIGENCE_OBJECT_MOCK,
+  OVERSEAS_ENTITY_DUE_DILIGENCE_REQ_BODY_OBJECT_MOCK_WITH_EMAIL_CONTAINING_LEADING_AND_TRAILING_SPACES,
+  OVERSEAS_ENTITY_DUE_DILIGENCE_REQ_BODY_EMPTY_OBJECT_MOCK,
+  OVERSEAS_ENTITY_DUE_DILIGENCE_REQ_BODY_MAX_LENGTH_FIELDS_MOCK,
+  OVERSEAS_ENTITY_DUE_DILIGENCE_REQ_BODY_OBJECT_MOCK
+} from "../../__mocks__/overseas.entity.due.diligence.mock";
+import {
+  WHO_IS_MAKING_UPDATE_URL,
+  UPDATE_DUE_DILIGENCE_OVERSEAS_ENTITY_PAGE,
+  UPDATE_DUE_DILIGENCE_OVERSEAS_ENTITY_URL,
+  OVERSEAS_ENTITY_REVIEW_PAGE,
+  OVERSEAS_ENTITY_REVIEW_URL
+} from "../../../src/config";
+import {
+  ANY_MESSAGE_ERROR,
+  SERVICE_UNAVAILABLE,
+  OVERSEAS_ENTITY_DUE_DILIGENCE_PAGE_TITLE,
+  OVERSEAS_ENTITY_DUE_DILIGENCE_NAME_TEXT,
+  FOUND_REDIRECT_TO,
+  OVERSEAS_ENTITY_NO_EMAIL_OR_VERIFICATION_DATE_SHOWN_INFORMATION_ON_PUBLIC_REGISTER,
+  PAGE_TITLE_ERROR,
+  OVERSEAS_ENTITY_DUE_DILIGENCE_IDENTITY_ADDRESS_HINT_TEXT,
+  OVERSEAS_ENTITY_DUE_DILIGENCE_PARTNER_NAME_HINT_TEXT,
+  OVERSEAS_ENTITY_DUE_DILIGENCE_IDENTITY_DATE_LABEL_TEXT,
+  OVERSEAS_ENTITY_DUE_DILIGENCE_PARTNER_NAME_LABEL_TEXT,
+  OVERSEAS_ENTITY_DUE_DILIGENCE_SUPERVISORY_NAME_LABEL_TEXT,
+  ALL_THE_OTHER_INFORMATION_ON_PUBLIC_REGISTER,
+} from "../../__mocks__/text.mock";
+
+const mockGetApplicationData = getApplicationData as jest.Mock;
+const mockSetApplicationData = setApplicationData as jest.Mock;
+const mockAuthenticationMiddleware = authentication as jest.Mock;
+
+const mockHasWhoIsMakingUpdate = hasWhoIsMakingUpdate as jest.Mock;
+mockHasWhoIsMakingUpdate.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockPrepareData = prepareData as jest.Mock;
+mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockServiceAvailabilityMiddleware = serviceAvailabilityMiddleware as jest.Mock;
+mockServiceAvailabilityMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+describe("UPDATE_DUE_DILIGENCE_OVERSEAS_ENTITY controller", () => {
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockSetApplicationData.mockReset();
+  });
+
+  describe("GET tests", () => {
+
```

Comment:
Remove empty line, I know this looks nicer but just trying to keep it consistent with the other test files